sklearn/linear_model/least_angle.py

```diff
@@ -48,22 +47,266 @@ def lars_path(X, y, Xy=None, Gram=None, max_iter=500,
+        Minimum correlation along the path. It corresponds to the
+        regularization parameter alpha parameter in the Lasso.
+
+    method : {'lar', 'lasso'}, optional (default='lar')
+        Specifies the returned model. Select ``'lar'`` for Least Angle
+        Regression, ``'lasso'`` for the Lasso.
+
+    copy_X : bool, optional (default=True)
+        If ``False``, ``X`` is overwritten.
+
+    eps : float, optional (default=``np.finfo(np.float).eps``)
+        The machine-precision regularization in the computation of the
+        Cholesky diagonal factors. Increase this for very ill-conditioned
+        systems.
+
+    copy_Gram : bool, optional (default=True)
+        If ``False``, ``Gram`` is overwritten.
+
+    verbose : int (default=0)
+        Controls output verbosity.
+
+    return_path : bool, optional (default=True)
+        If ``return_path==True`` returns the entire path, else returns only the
+        last point of the path.
+
+    return_n_iter : bool, optional (default=False)
+        Whether to return the number of iterations.
+
+    positive : boolean (default=False)
+        Restrict coefficients to be >= 0.
+        This option is only allowed with method 'lasso'. Note that the model
+        coefficients will not converge to the ordinary-least-squares solution
+        for small values of alpha. Only coefficients up to the smallest alpha
+        value (``alphas_[alphas_ > 0.].min()`` when fit_path=True) reached by
+        the stepwise Lars-Lasso algorithm are typically in congruence with the
+        solution of the coordinate descent lasso_path function.
+
+    Returns
+    --------
+    alphas : array, shape (n_alphas + 1,)
+        Maximum of covariances (in absolute value) at each iteration.
+        ``n_alphas`` is either ``max_iter``, ``n_features`` or the
+        number of nodes in the path with ``alpha >= alpha_min``, whichever
+        is smaller.
+
+    active : array, shape [n_alphas]
+        Indices of active variables at the end of the path.
+
+    coefs : array, shape (n_features, n_alphas + 1)
+        Coefficients along the path
+
+    n_iter : int
+        Number of iterations run. Returned only if return_n_iter is set
+        to True.
+
+    See also
+    --------
+    lasso_path
+    lasso_path_gram
+    LassoLars
+    Lars
+    LassoLarsCV
+    LarsCV
+    sklearn.decomposition.sparse_encode
+
+    References
+    ----------
+    .. [1] "Least Angle Regression", Efron et al.
+           http://statweb.stanford.edu/~tibs/ftp/lars.pdf
+
+    .. [2] `Wikipedia entry on the Least-angle regression
+           <https://en.wikipedia.org/wiki/Least-angle_regression>`_
+
+    .. [3] `Wikipedia entry on the Lasso
+           <https://en.wikipedia.org/wiki/Lasso_(statistics)>`_
+
+    """
+    if X is None and Gram is None:
+        warnings.warn('Use lars_path_gram to avoid passing X and y. '
+                      'The current option will be removed in future versions.',
```

Comment:
which current option? the Gram parameter? We need to specify which future version (0.23, I'd think).

the `Gram is None` condition here seems weird. Should it be `Gram is not None`?

Reply:
Fixed.